### PR TITLE
Stylelint

### PR DESCRIPTION
**Pull request by amcclain**
Hoist P/R Checklist
-------------------

Review and check off the below. Items that do not apply can also be checked off to indicate they
have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or open as a draft PR / add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [NA] Reviewed for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
I don't necessarily feel that the stylelint default choices are strictly better than what we had in place, but also was happy to take them as standards and just roll with it.  I think having consistency is worth it more than anything. Hopefully you don't mind the changes the linter made to the styles and taking these new conventions going forward.

I did disable the `no-descending-specificity` rule as it seemed overly fussy to me and there were a decent number of breaks (and no auto-fix).  This rule says that if you have two selectors targeting some class or element, they should be listed in the file in order of increasing specificity. I can't say it seemed to add much, and some of the changes made the rules less readable to me.  Happy of course to tweak other rules.

If we decide we want to use this more widely, we could publish a Hoist stylelint ruleset as a package - just as we do with https://github.com/xh/eslint-config - but I wasn't in any rush to jump to that. 

Opened up related, although not actually dependent, PR on Toolbox @ https://github.com/xh/toolbox/pull/282

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "plugins": [
+        "stylelint-scss"
+    ],
+    "extends": [
+        "stylelint-config-standard"
+    ],
+    "rules": {
+        "at-rule-no-unknown": null,
```
Comment (amcclain) [inline .stylelintrc]: We use some SASS-specific directives that are flagged as unrecognized by stylelint out of the box - they made the deliberate decision to only include pure CSS @ rules.  So this rule is turned off here, and instead we use the stylelint-scss plugin and enable the `scss/at-rule-no-unknown' rule it provides below to offer the same coverage, just with built-in awareness of standard sass functions, etc.

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        "no-descending-specificity": null,
+        "property-no-unknown": [
+            true,
+            {"ignoreSelectors": [":export"]}
+        ],
+        "scss/at-rule-no-unknown": true,
+        "selector-pseudo-class-no-unknown": [
+            true,
+            {"ignorePseudoClasses": ["export"]}
```
Comment (amcclain) [inline .stylelintrc]: The two rule customizations related to `export` cover an edge-case usage we have in https://github.com/xh/hoist-react/blob/develop/desktop/cmp/grid/columns/Actions.scss where we do some fancy stuff to make a value importable from JS.

**Comment by TomTirapani** [comment]
Overall, looks great - and like you say, the main benefit here is to get some kind of standard in place.

The only thing I can think of is to perhaps set `except: ["after-single-line-comment"]` for this rule: https://stylelint.io/user-guide/rules/rule-empty-line-before . I find the new empty lines between comments and the style makes it a bit less clear what the comment refers to.

**Pull request merged**